lib/trivial.nix

```diff
@@ -53,6 +53,30 @@ rec {
   # argument, but it's nice this way if several uses of `extends` are cascaded.
   extends = f: rattrs: self: let super = rattrs self; in super // f self super;
 
+  # Create an overridable, recursive attribute set. For example:
+  #
+  #     nix-repl> obj = makeObject (self: { })
+  #
+  #     nix-repl> obj
+  #     { override = «lambda»; }
+  #
+  #     nix-repl> obj = obj.override (self: super: { foo = "foo"; })
+  #
+  #     nix-repl> obj
+  #     { foo = "foo"; override = «lambda»; }
+  #
+  #     nix-repl> obj = obj.override (self: super: { foo = super.foo + " + "; bar = "bar"; foobar = self.foo + self.bar; })
+  #
+  #     nix-repl> obj
+  #     { bar = "bar"; foo = "foo + "; foobar = "foo + bar"; override = «lambda»; }
+  makeObject = rattrs:
+    let
+      construct = rattrs:
+        fix (self: rattrs self // {
+          override = f: construct (extends f rattrs);
+        });
+    in construct rattrs;
```

Comment:
@cstrahan I'd really like to use this, but let's first simplify as @roconnor suggested :)